torchvision/transforms/functional_tensor.py

```diff
@@ -44,22 +44,14 @@ def get_image_num_channels(img: Tensor) -> int:
     raise TypeError(f"Input ndim should be 2 or more. Got {img.ndim}")
 
 
-def _max_value(dtype: torch.dtype) -> float:
-    # TODO: replace this method with torch.iinfo when it gets torchscript support.
-    # https://github.com/pytorch/pytorch/issues/41492
-
-    a = torch.tensor(2, dtype=dtype)
-    signed = 1 if torch.tensor(0, dtype=dtype).is_signed() else 0
-    bits = 1
-    max_value = torch.tensor(-signed, dtype=torch.long)
-    while True:
-        next_value = a.pow(bits - signed).sub(1)
-        if next_value > max_value:
-            max_value = next_value
-            bits *= 2
-        else:
-            break
-    return max_value.item()
+def _max_value(dtype: torch.dtype) -> int:
+    return {
+        torch.uint8: int(2 ** 8) - 1,
+        torch.int8: int(2 ** 7) - 1,
+        torch.int16: int(2 ** 15) - 1,
+        torch.int32: int(2 ** 31) - 1,
+        torch.int64: int(2 ** 63) - 1,
+    }.get(dtype, 1)
```

Comment:
> Given that this method is being called often, does it make sense to move the dictionary definition on global scope to avoid reinitialisation?

```py
import torch
from typing import Dict

_MAX_VALUES: Dict[torch.dtype, int] = {
    torch.uint8: int(2 ** 8) - 1,
    torch.int8: int(2 ** 7) - 1,
    torch.int16: int(2 ** 15) - 1,
    torch.int32: int(2 ** 31) - 1,
    torch.int64: int(2 ** 63) - 1,
}


@torch.jit.script
def max_value_global(dtype: torch.dtype) -> int:
    return _MAX_VALUES.get(dtype, 1)
```

fails with

```
RuntimeError: 
attribute lookup is not defined on python value of type 'dict'
```

I also recall Victor saying that globals are not supported.

> If that causes JIT issues, we could also do ifs.

Although the timings are hardly significant here, it seems that the dictionary approach is slightly faster:

```py
import torch
import itertools
from time import perf_counter_ns


@torch.jit.script
def max_value_dict(dtype: torch.dtype) -> int:
    return {
        torch.uint8: int(2 ** 8) - 1,
        torch.int8: int(2 ** 7) - 1,
        torch.int16: int(2 ** 15) - 1,
        torch.int32: int(2 ** 31) - 1,
        torch.int64: int(2 ** 63) - 1,
    }.get(dtype, 1)


@torch.jit.script
def max_value_if(dtype: torch.dtype) -> int:
    if dtype == torch.uint8:
        return int(2 ** 8) - 1
    elif dtype == torch.int8:
        return int(2 ** 7) - 1
    elif dtype == torch.int16:
        return int(2 ** 15) - 1
    elif dtype == torch.int32:
        return int(2 ** 31) - 1
    elif dtype == torch.int64:
        return int(2 ** 63) - 1
    else:
        return 1


num = 1_000_000

dict_timings = []
if_timings = []

for dtype in itertools.islice(
    itertools.cycle(torch.testing._legacy.get_all_dtypes(include_bool=False, include_complex=False)), num
):
    tic = perf_counter_ns()
    max_value_dict(dtype)
    tac = perf_counter_ns()
    dict_timings.append(tac - tic)

    tic = perf_counter_ns()
    max_value_if(dtype)
    tac = perf_counter_ns()
    if_timings.append(tac - tic)

print(f"{torch.tensor(dict_timings, dtype=torch.float64).mean() * 1e-3:.3f} µs per call using dicts")
print(f"{torch.tensor(if_timings, dtype=torch.float64).mean() * 1e-3:.3f} µs per call using ifs")
```

```
0.990 µs per call using dicts
1.100 µs per call using ifs
```

Reply:
In eager mode ifs are indeed faster

```
0.565 µs per call using dicts
0.267 µs per call using ifs
```

but here the timings are becoming even more insignificant.

Reply:
:) I think it is overthinking but OK i'll change it to if/else implementation if it is better